packages/compass-aggregations/src/modules/pipeline-builder/text-editor-output-stage.ts

```diff
@@ -0,0 +1,167 @@
+import type { Reducer } from 'redux';
+import type { AggregateOptions, MongoServerError } from 'mongodb';
+import { globalAppRegistryEmit } from '@mongodb-js/mongodb-redux-common/app-registry';
+import type { PipelineBuilderThunkAction } from '..';
+import { DEFAULT_MAX_TIME_MS } from '../../constants';
+import { isAction } from '../../utils/is-action';
+import { EditorActionTypes, canRunPipeline } from './text-editor';
+import type { EditorValueChangeAction } from './text-editor';
+
+import { CONFIRM_NEW, NEW_PIPELINE } from '../import-pipeline';
+import { RESTORE_PIPELINE } from '../saved-pipeline';
+import { aggregatePipeline } from '../../utils/cancellable-aggregation';
+import { gotoOutResults } from '../out-results-fn';
+
+export const enum OutputStageActionTypes {
+  FetchStarted = 'compass-aggregations/pipeline-builder/text-editor-output-stage/FetchStarted',
+  FetchSucceded = 'compass-aggregations/pipeline-builder/text-editor-output-stage/FetchSucceded',
+  FetchFailed = 'compass-aggregations/pipeline-builder/text-editor-output-stage/FetchFailed',
+};
+
+type OutputStageFetchStartedAction = {
+  type: OutputStageActionTypes.FetchStarted;
+};
+
+type OutputStageFetchSuccededAction = {
+  type: OutputStageActionTypes.FetchSucceded;
+};
+
+type OutputStageFetchFailedAction = {
+  type: OutputStageActionTypes.FetchFailed;
+  serverError: MongoServerError;
+};
+
+export type OutputStageState = {
+  isLoading: boolean,
+  serverError: MongoServerError | null,
+  isComplete: boolean;
+};
+
+const INITIAL_STATE: OutputStageState = {
+  isLoading: false,
+  isComplete: false,
+  serverError: null,
```

Comment:
Based on usage this is the same as serverError in text editor slice, why are we keeping this one separate?